### PR TITLE
refactor: Move existing drawing to ECS style system

### LINKED ISSUE #1018: TileMap should have getZIndex() too
There is currently no straight way to draw TileMap other than in background. Would be better if they'd have getZIndex() property, like Actor, and be drawn accordingly

### PR DESCRIPTION
**Pull request by eonarheim**
===:clipboard: PR Checklist :clipboard:===

- [x] :pushpin: issue exists in github for these changes
- [x] :microscope: existing tests still pass
- [x] :see_no_evil: code conforms to the [style guide](https://github.com/excaliburjs/Excalibur/blob/main/STYLEGUIDE.md)
- [x] :triangular_ruler: new tests written and passing / old tests updated with new scenario(s)
- [x] :page_facing_up: changelog entry added (or not needed)

==================

This change comes with a lot of advantages, everything including TileMaps can make use of z-indexing, anything with a pos, rotation, scale, and draw() method can be drawn.

Related #1361
Closes #1018 #888 

Docs PR: https://github.com/excaliburjs/excaliburjs.github.io/pull/57

## Changes:

- Moves existing drawing functionality to ECS
- Updates to ECS implementation to facilitate Excalibur integration

### CODE REVIEW

File: src/engine/Drawing/CanvasDrawingSystem.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      this._ctx.save();
+      if (this._camera) {
+        this._camera.draw(this._ctx);
+      }
+    }
+  }
+
+  private _popCameraTransform(transform: TransformComponent) {
+    if (transform.coordPlane === CoordPlane.World) {
+      // Apply camera world offset
```
Comment (jedeen) [inline src/engine/Drawing/CanvasDrawingSystem.ts:103]: is this applying the world offset to the camera, or applying a camera offset to the world?

File: src/engine/EntityComponentSystem/System.ts [hunk 2]
```diff
@@ -27,31 +52,44 @@ export abstract class System<T extends Component = Component> implements Observe
    */
   public priority: number = 0;
 
+  /**
+   * Optionally Sepecify a sort order for entities passed to the your system
```
Comment (jedeen) [inline src/engine/EntityComponentSystem/System.ts:56]:
```suggestion
   * Optionally specify a sort order for entities passed to the your system
```

File: src/engine/EntityComponentSystem/World.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
+import { Entity } from './Entity';
+import { EntityManager } from './EntityManager';
+import { QueryManager } from './QueryManager';
+import { System, SystemType } from './System';
+import { SystemManager } from './SystemManager';
+
+/**
+ * The World is a self contained entity component system for a particular context.
```
Comment (jedeen) [inline src/engine/EntityComponentSystem/World.ts:8]:
```suggestion
 * The World is a self-contained entity component system for a particular context.
```

**Review by jedeen — CHANGES_REQUESTED** [review]

File: src/engine/Drawing/CanvasDrawingSystem.ts (hunk 1 above)
Comment (eonarheim) [reply src/engine/Drawing/CanvasDrawingSystem.ts:103]: Good question, `_pushCameraTransform` applies the camera's transform to the graphics context which is world space. This comment here seems out of place in `_popCameraTransfrom` which restores the context back to screen space.

Comment (eonarheim) [reply src/engine/Drawing/CanvasDrawingSystem.ts]: I've updated the comments to give more clarity

**Review by jedeen — APPROVED** [review]

**Pull request merged**